### DISCUSSION TITLE
[ShapeOptimization] json file contents

### DISCUSSION THREAD

**Discussion by Ada2045lalala** [KratosMultiphysics/Examples#134] [Q&A]
1.I don't know many about the Kratos, and just come across [this example](https://github.com/KratosMultiphysics/Examples/tree/master/shape_optimization/use_cases/11_Shape_Update_Optimization_Stanford_Bunny),which is similar to my research. Therefore, I tried to use the [ShapeOptimizationApplication](https://github.com/KratosMultiphysics/Kratos/tree/master/applications/ShapeOptimizationApplication) to solve my optimization.
2.Now, the json file confuses me. (I've read [this tutorial](https://github.com/KratosMultiphysics/Kratos/wiki/How-to-write-a-JSON-configuration-file),unfortunately,it fails to answer my doubts). Some items are easy to understand by their names, like domain_size, input_type.
![json](https://github.com/KratosMultiphysics/Examples/assets/155991150/30cfab74-6476-4496-9904-7407679c93ee)
however, most of the items are far beyond my knowledge. like these: 
![屏幕截图 2024-02-28 210531](https://github.com/KratosMultiphysics/Examples/assets/155991150/6471cd7f-b81d-46fb-a3a9-93dcace917b8)
and so on.
3.My problem is how could a user like me understand the items as above, have you provided any relevant documents which I ignored? Or I need to read all the codes in Kratos and its applications to figure it out by myself?

I will really appreciate you if you could so kindly do me a favor!

**Comment by rubenzorrilla** [comment]
I think @sunethwarna is the best person to answer this.

**Comment by sunethwarna** [comment]
Hi @Ada2045lalala,

It is really nice to see that you are interested in using `ShapeOptimization` application. We have a documentation, but not yet in the master [We are working on more general optimization application called `OptimizationApplication`, but it is not yet ready for production, we have to do testing]. So if you are planning to use `ShapeOpt` app, then I would recommend the following link for documentation (**go inside each folder and read the markdown (*.md) files in github, they should render the markdown files**):

https://github.com/KratosMultiphysics/Kratos/tree/docs/shapeopt/docs/pages/Applications/Shape_Optimization_Application

Other than that, you can either go through the tests under the folder:
`applications/ShapeOptimizationApplication/tests`, or have to go through the examples.

If you have some specific questions, then I would be really glad to help you out. :+1: 

Best regards,
Suneth.
